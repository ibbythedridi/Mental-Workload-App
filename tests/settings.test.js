import React from 'react';
import renderer from 'react-test-renderer';
import Settings from '../screens/settings';

it('renders settings correctly', () => {
  const tree = renderer.create(<Settings />).toJSON();
  expect(tree).toMatchSnapshot();
});

// This checks the correct component(s) are existent in the JSON
describe("<Settings />", () => {
  it('has 2 children', () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});