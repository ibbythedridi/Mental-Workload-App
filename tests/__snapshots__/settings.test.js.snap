// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders settings correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f2f3f7",
        "borderRadius": 7,
        "elevation": 3,
        "marginVertical": 10,
        "padding": 10,
        "shadowColor": "#222",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Developer Options
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#0043ff",
            "borderRadius": 7,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          Add Example Data
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 48,
          },
        ],
        "zIndex": 99,
      }
    }
  />
</View>
`;
